### DISCUSSION TITLE
Qual seu primeiro projeto opensource?

### DISCUSSION THREAD

**Discussion by willianjusten** [frontendbr/forum#164] [Discussão]
Fala pessoal, hoje fiquei bem feliz com uma conquista, que foi ter tido o commit de número 400 num dos meus maiores projetos do Github que é a [awesome-svg](https://github.com/willianjusten/awesome-svg).

O projeto nasceu de uma vontade de expôr os vários conteúdos que eu tinha, achei que um ou outro iria acessar, mas fez mais sucesso que jamais imaginei.

E você? Qual foi seu primeiro projeto ou o que você considera mais importante? E como ele nasceu?

**Comment by woliveiras** [comment]
Meu projeto mais antigo foi o [Pastel](https://github.com/woliveiras/pastel), um Grunt Boilerplate com as tarefas que usava no desenvolvimento de sites institucionais.

Ele nasceu para agilizar o início do desenvolvimento dos Sites em uma agência que trabalhei. - Por isso o nome Pastel (os clientes achavam que se faz site fritando em 5 minutinhos)

Ele também foi bem importante pra mim, pois usei várias vezes e conheci bastante gente graças a ele (mesmo com poucas estrelinhas kkk).

É esse projeto, também, que marca quanto tempo eu tenho de carreira como Desenvolvedor Front End!

**Comment by theandersonn** [comment]
Meu primeiro projeto é o "[Comandos Git](https://github.com/theandersonn/comandos-git)", uma coletânea com o meu aprendizado com Git e GitHub.

Bom, eu tive muita dificuldade em aprender a utilizar o Git, não trabalhava com terminal e perdia a paciência com os conflitos; desisti algumas vezes e num determinado momento fui pro tudo ou nada, comecei a escrever os comandos em papéis e cola-los em volta do monitor :) Foi daí que nasceu o repositório.

De lá pra cá... já palestrei duas vezes sobre o assunto, consegui ajudar alguns amigos com a ferramenta e aprendi bastante sobre o temido terminal, consegui migrar definitivamente 
para Linux. \o/

**Comment by afonsopacifer** [comment]
O meu foi o [awesome-flexbox](https://github.com/afonsopacifer/awesome-flexbox), diria que pelos mesmos motivos do @willianjusten :) (que por sinal foi a inspiração para eu criar o projeto)

Uma curiosidade é que terminei ele durante a CSS Conf Brasil (na hora do almoço) e isso foi no dia do meu aniversário tbm (q loko) haha

**Comment by Pompeu** [comment]
bahh meu projeto foi [https://github.com/pompeu/rede](Rede de conhecimentos),  onde fiz um prototipo de uma rede social com NodeJS , Angular e Neo4j, durante minha inciação cientifica na faculdade.
Ainda não esta finalizada mais esse ano termino e vou gerar contatos de conhecimentos na academia :P e sera sempre opensource.

**Comment by tfvictorino** [comment]
O meu é o [modular-ui-directory](https://github.com/modular-ui/modular-ui-directory). Quando comecei a estudar sobre arquitetura de CSS há uns 2 anos atrás para um projeto, fui encontrando muito material interessante e que apontava para outros conceitos relacionados, como design systems, modularidade, patterns libraries... Vi que era uma nova forma de se pensar e projetar interfaces (no âmbito do design e do front-end) e me identifiquei bastante com o que a galera estava propondo. 

Fui salvando nos favoritos, mas teve uma hora que virou o caos, haha. Nesse meio-tempo, um amigo meu estava interessado em estudar os mesmos assuntos, então vi que era hora de organizar de alguma forma que ficasse acessível pra mais gente. Aí saiu o repositório! Acabou motivando também a criação do [ModularUI](http://www.modular-ui.com), iniciativa que promove discussões sobre esses assuntos. 😃

**Comment by Marcelosilva10** [comment]
Só eu que ainda não tenho nenhum projeto preciso ter algo em mente também.

**Comment by woliveiras** [comment]
Não se force a criar algo só pra ter um projeto e "parecer" com a maioria @Marcelosilva10.

Uma coisa que você consegue perceber é que todos que criaram algo, fizeram pra sanar algum problema ou para melhorar algo par si e acabaram ajudando mais gente.

Talvez esteja ai o por que de ainda não ter algo.

Se você se forçar a ter uma ideia só pra ter um projeto pra compartilhar nas redes sociais, você estará desperdiçando sua energia, sendo que, a maioria das vezes, você já deve ter algum problema que pode resolver e compartilhar. kkkk

Crie algo pra você. Se for útil, compartilhe com todo mundo, do mundo todo, pra todo mundo. \o/

Se ainda não tem nada do tipo, gaste energia estudando e praticando mais, compartilhe o que estuda, ajude alguém, etc. :smile: 

Você também pode colaborar nos projetos, cara. Acho mais útil do que forçar uma ideia. Forçando fica parecendo aqueles empresários que ficam criando 700 Startups com ideias forçadas que, quase sempre, não dão certo por não serem úteis de verdade.

**OBS 1**: todos os meus projetos aceitam Pull Request ein...  :laughing:

**OBS 2**: Não se sinta repreendido ein cara, é só uma dica de amigo que já passou por algumas dessas 700 Startups com ideias forçadas. kkkkkk

**Comment by felipefialho** [comment]
O primeiro foi um boilerplate de [Grunt, em meados de 2013](https://github.com/LFeh/grunt-boilerplate).

Mas acho que projeto mesmo, foi o [Piano Keyboard](https://github.com/LFeh/piano-keyboard).

:smile_cat:

**Comment by jdrgoms** [comment]
Fala galera, beleza? Então, meu primeiro projeto (  [****awesome-postcss****](https://github.com/jjaderg/awesome-postcss) ) surgiu depois de uma palestra que fiz num meetup, e também foi inspirado por outros dois projetos de amigos  @afonsopacifer & @willianjusten :grin:

**Comment by fdaciuk** [comment]
Meu primeiro projeto open source foi o [da2k-grid](https://github.com/fdaciuk/da2k-grid), em uma época em que bootstrap e similares começava a reinar, e não existia nenhuma solução de grid decente pra usar somente com CSS. Pouco tempo depois surgiram uns grids legais que não precisavam adicionar classe no HTML, usando a mesma pegada do que eu tava começando, ae acabei largando pra lá xD

Mas foi uma experiência legal :grin: 

Depois comecei o [sexyforms](https://github.com/fdaciuk/sexyforms), pra estilizar campos de formulário de forma simples (especificamente selects, radio e checkbox), mas usava jQuery ainda. Acabei não levando pra frente também =)

Hoje o meu xodó é o [@fdaciuk/ajax](https://github.com/fdaciuk/ajax) (que vai ficar obsoleto em breve, com a chegada do `fetch` :P), mas eu gosto bastante dele, e vou tentar manter por um bom tempo. Quem sabe no futuro ele não utilize a API do fetch, ou quando o fetch já estiver presente em todos os browsers, essa lib possa ficar como legado xD

Tem alguns outros projetos que eu mantenho também: [hoost](https://github.com/fdaciuk/hoost) para gerenciar seus virtual hosts, [hmh-cli](https://github.com/fdaciuk/hmh-cli), pra fazer cálculo de horas direto no terminal, [boleto](https://github.com/fdaciuk/boleto) pra atualizar datas de boletos quando preciso pagar contas, e o [iscpf](https://github.com/fdaciuk/iscpf), pra checar se o CPF é válido xD

Acho que me empolguei :stuck_out_tongue_winking_eye:

**Comment by cjpatoilo** [comment]
Meu primeiro projeto open source foi o [Milligram](https://github.com/milligram/milligram). 

Bom, o primeiro deveria ter sido outro projeto ao invés do Milligram. Algo bastante parecido com a necessidade descrita pelo @woliveiras. Eu já havia produzido a API e a documentação. Faltava apenas desenvolver a homepage, daí pensei: _"Po, eu poderia utilizar meu próprio Framework"_. 

Então, eu já possuia essa ideia há bastante tempo. Resolvi mudar o foco e dedicar todo o meu tempo livre para produzir esse projeto. Toda essa experiência foi incrível, aprendi muito mais do que eu esperava. =D

**Comment by Marcelosilva10** [comment]
@woliveiras podeixar vou ver alguns la que posso ajudar, obrigado!

**Comment by ericktatsui** [comment]
Revivendo essa issue, gostaria de mostrar meu primeiro projeto open source.
Foi um [touch menu inspirado no nativo do android](https://github.com/ericktatsui/Touch-Menu-Like-Android) para ser usado em aplicações mobile hibridas ou até mesmo na web, eu usei a biblioteca hammer.js para as interações touch, se puderem dar um feedback, da uma olhada no código ou alguma dica eu agradeço :D

**Comment by felipefialho** [comment]
Parabéns @ericktatsui ;)

**Comment by kandebonfim** [comment]
O meu primeiro foi recente. Sempre achei massa mas nunca tinha metido a cara. Criei o [itcsscli](http://www.itcsscli.com/), uma rubygem pra ajudar quem usa ITCSS no frontend.

**Comment by akfzambrana** [comment]
Acho que meu primeiro "projeto" mesmo foi a minha [vagrant box](https://github.com/akfzambrana/vagrant-box) que ultimamente ta dando uns paus bizarros, mas tamos ai, de pouco em pouco acabo mexendo nela kkkk
A dica do @woliveiras pro @Marcelosilva10 é muito util mesmo, começa por fazer algo pra vc e qdo for ver alguem acaba curtindo, eu mesmo mantenho no meu github apenas trechos de coisas q são úteis pra mim. Antigamente não tinha nem isso :wink:

**Comment by douglascorrea** [comment]
Descobri esse repo agora e estou achando muito bom! Parabéns pro pessoal que mantém esse projeto, também acompanho o repo de vagas!

O meu primeiro projeto open source foi o [angularfire-multi-auth](https://github.com/douglascorrea/angularfire-multi-auth) que foi uma tentativa de suprir uma necessidade antiga que o Firebase não suportava: Múltiplos Social providers na mesma conta. 

Mais recentemente eu fiz meu segundo projeto open source, também ligado ao Firebase, que teve maior repercussão: [Um Boilerplate para Firebase 3.0 com React/Redux](https://github.com/douglascorrea/react-hot-redux-firebase-starter) e o pessoal tem dado bons feedbacks e conversado bastante comigo no Slack do Firebase.

**Comment by fakenickels** [comment]
Revivendo um pouco a issue,
Meu primeiro código-aberto foi o https://github.com/grsabreu/Leran/, uma Ruby gem bem chinfrim que organiza arquivos pela sua extensão.
Mas o que eu realmente considero primeiro é o https://github.com/grsabreu/Animelt-Plugin, um motor de animações que funciona como um fallback pras do CSS3 que fiz com a ajuda de um amigo. Foi nele que eu me apaixonei por JS e meu primeiro "massaaa" depois de ver meu bagulho massa funcionando.
O projeto tá morto, mas olho de vez em quando pra ver o quanto eu evoluí de lá pra cá.

**Comment by barbier** [comment]
(Nem acredito que estou fazendo isso...)
Eu acabei de publicar uma [calculadora de quantidade de terrenos de Magic the Gathering.](https://github.com/barbier/mana-calc)
